src/Albany_StateManager.cpp

```diff
@@ -62,9 +62,9 @@ StateManager::registerStateVariable(
   Teuchos::RCP<Teuchos::ParameterList> p =
       Teuchos::rcp(new Teuchos::ParameterList(
           "Save or Load State " + stateName + " to/from field " + stateName));
-  p->set<const std::string>("State Name", stateName);
-  p->set<const std::string>("Field Name", stateName);
-  p->set<const Teuchos::RCP<PHX::DataLayout>>("State Field Layout", dl);
```

Comment:
Yea, it definingly did not make sense to use `const` with this template parameter to the PL::set() function.  (The value is already passed in as `const T&` so there was no need to specify `T` as `const`, which would make `const const std::string&`?)

Now, if you wanted to be sure to control the type used to store the parameter value, you could have just dropped `const` and called:

```
p->set<std::string>("State Name", stateName);
```

That makes 100% sense.

I am kind of surprised this compiled even with the old implementation for PL::set().  (C++ templates are so complex.)